cleanlab/internal/outlier.py

```diff
@@ -33,18 +35,17 @@ def transform_distances_to_scores(distances: np.ndarray, k: int, t: int) -> np.n
 
     Parameters
     ----------
-    distances : np.ndarray
-        An array of distances of shape ``(N, num_neighbors)``, where N is the number of examples.
-        Each row contains the distances to each example's `num_neighbors` nearest neighbors.
-        It is assumed that each row is sorted in ascending order.
-
-    k : int
-        Number of neighbors used to compute the average distance to each example.
-        This assumes that the second dimension of distances is k or greater, but it
-        uses slicing to avoid indexing errors.
+    avg_distances : np.ndarray
+        An array of distances of shape ``(N)``, where N is the number of examples.
+        Each row represents an example's average distance to its k nearest neighbors.
 
     t : int
-        Controls transformation of distances between examples into similarity scores that lie in [0,1].
+        A sensitivity parameter that modulates the strength of the transformation from distances to scores.
+        Higher values of `t` result in more pronounced differentiation between the scores of examples
+        lying in the range [0,1].
+
+    scaling_factor : float
```

Comment:
What is the range for this/what should it be ideally? What is the repercussion of making this value larger or smaller?

Reply:
A valid scaling factor lies in the range $(0,  +\infty)$, but ideally, it should be selected based on some reference distance measured in the dataset. By default, its value is set to the median of the average distances to the k-nearest neighbors. The choice of scaling factor is flexible.

- The value can be increased (_decreased_) to dilute (_amplify_) the relative differences in distances between examples (the distribution of scores will be shifted towards 1 (_0_)).